lib/change_stream.js

```diff
@@ -281,25 +281,36 @@ var createChangeStreamCursor = function(self) {
   return changeStreamCursor;
 };
 
+function getResumeToken(self) {
+  return self.resumeToken || self.options.resumeAfter;
+}
+
+function getStartAtOperationTime(self) {
+  const isMaster = self.serverConfig.lastIsMaster() || {};
+  return (
+    isMaster.maxWireVersion && isMaster.maxWireVersion >= 7 && self.options.startAtOperationTime
+  );
+}
+
 var buildChangeStreamAggregationCommand = function(self) {
   const serverConfig = self.serverConfig;
   const namespace = self.namespace;
   const pipeline = self.pipeline;
-  const resumeToken = self.resumeToken;
   const options = self.options;
   const cursorNamespace = self.cursorNamespace;
 
-  const isMaster = serverConfig.lastIsMaster() || {};
-
   var changeStreamStageOptions = {
     fullDocument: options.fullDocument || 'default'
   };
 
-  if (resumeToken || options.resumeAfter) {
-    changeStreamStageOptions.resumeAfter = resumeToken || options.resumeAfter;
-  } else if (isMaster.maxWireVersion && isMaster.maxWireVersion >= 7) {
-    const ts = options.startAtOperationTime || self.operationTime;
-    changeStreamStageOptions.startAtOperationTime = ts;
+  const resumeToken = getResumeToken(self);
+  const startAtOperationTime = getStartAtOperationTime(self);
+  if (resumeToken) {
+    changeStreamStageOptions.resumeAfter = resumeToken;
+  }
+
+  if (startAtOperationTime) {
+    changeStreamStageOptions.startAtOperationTime = options.startAtOperationTime;
```

Comment:
looks like this should be `startAtOperationTime` not `options. startAtOperationTime`

Reply:
yes, it should, thanks!!!